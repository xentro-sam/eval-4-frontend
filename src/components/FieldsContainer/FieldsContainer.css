#fields-container-body {
    padding: 2vw;
    color: rgb(79, 79, 79);
}

#fields-container-header {
    font-size: 3vw;
    font-weight: 600;
    display: flex;
    justify-content: flex-end;
}

#fields-count {
    font-size: 2vw;
    font-weight: 300;
    padding: 1vw 0;
}

#container-title-edit > img {
    width: 2vw;
    height: 2vw;
    margin: 0 1vw;
    cursor: pointer;
}

#add-field-button {
    border: 2px dotted rgb(78, 34, 198);
    color: rgb(78, 34, 198);
    font-size: 1vw;
    font-weight: 600;
    border-radius: 5px;
    padding: 1vw 2vw;
    background-color: rgb(233, 234, 241);
    text-align: center;
    margin: 1vw;
    cursor: pointer;
}