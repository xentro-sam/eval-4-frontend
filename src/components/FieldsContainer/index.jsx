import * as React from 'react';
import './FieldsContainer.css';
import PropTypes from 'prop-types';
import Fields from '../Fields';
import {v4 as uuidv4} from 'uuid';
import NewFieldModal from '../NewFieldModal';

export default function FieldsContainer(props) {
  const [show, setShow] = React.useState(false);
  return (
    <div id="fields-container-body">
      <div id="fields-container-header">
        {props.title}
      </div>
      <div id="fields-count">
        {props.fields.length} fields
      </div>
      <div id="add-field-button" onClick={() => setShow(true)}>
        Add Another Field
      </div>
      <NewFieldModal show={show} onClose={() => setShow(false)} contentTypeId={props.contentTypeId} setFields={props.setFields} />
      <div id="fields-container-content">
        {props.fields.map((field) => {
          return <Fields fieldName={field} fieldType={'Text'} key={uuidv4()} contentTypeId={props.contentTypeId} />;
        })
        }
      </div>
    </div>
  );
}

FieldsContainer.propTypes = {
  title: PropTypes.string.isRequired,
  fields: PropTypes.array,
  contentTypeId: PropTypes.number,
  setFields: PropTypes.func.isRequired,
};
