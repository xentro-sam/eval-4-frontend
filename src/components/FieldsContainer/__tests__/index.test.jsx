import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FieldsContainer from '../index';
import {waitFor} from '@testing-library/dom';
import makeRequest from '../../../utils/makeRequest';
import {act} from 'react-dom/test-utils';

jest.mock('../../../utils/makeRequest');

const mockData = {
  title: 'name',
  fields: ['name', 'age'],
  contentTypeId: 123,
  setFields: jest.fn(),
  setContainerTitle: jest.fn(),
};

describe('ContentType', () => {
  it('should render correctly', async () => {
    expect(makeRequest).not.toHaveBeenCalled();
    const {asFragment} = await act( async () => render(<FieldsContainer {...mockData} />));
    await waitFor(() => {
      expect(screen.getByText('Add Another Field')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
