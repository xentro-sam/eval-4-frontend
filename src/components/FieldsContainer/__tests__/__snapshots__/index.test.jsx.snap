// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    id="fields-container-body"
  >
    <div
      id="fields-container-header"
    >
      <div
        id="fields-container-title"
      >
        name
      </div>
      <div
        id="container-title-edit"
      >
        <img
          alt="container-title-edit"
          src="user-pencil-write-ui-education@3x.png"
        />
      </div>
    </div>
    <div
      id="fields-count"
    >
      2 fields
    </div>
    <div
      id="add-field-button"
    >
      Add Another Field
    </div>
    <div
      id="fields-container-content"
    >
      <div
        id="field-body"
      >
        <div
          id="field-name"
        >
          name
        </div>
        <div
          id="field-type"
        >
          Text
        </div>
        <div
          id="field-actions"
        >
          <div
            id="field-delete"
          >
            <img
              alt="delete-icon"
              src="trash-delete-recycle-bin-bucket-waste@3x.png"
            />
          </div>
          <div
            id="field-edit"
          >
            <img
              alt="edit-icon"
              src="user-pencil-write-ui-education@3x.png"
            />
          </div>
        </div>
      </div>
      <div
        id="field-body"
      >
        <div
          id="field-name"
        >
          age
        </div>
        <div
          id="field-type"
        >
          Text
        </div>
        <div
          id="field-actions"
        >
          <div
            id="field-delete"
          >
            <img
              alt="delete-icon"
              src="trash-delete-recycle-bin-bucket-waste@3x.png"
            />
          </div>
          <div
            id="field-edit"
          >
            <img
              alt="edit-icon"
              src="user-pencil-write-ui-education@3x.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
